### PR TITLE
Add ability to add source to charts

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

In order for a chart/viz to be taken serious, it must have the source of the data attributed.

This PR add ability to define source either as a link via `data-source` attribute on chart div (html) or as a `source` object in the `metadata` section of chart data (python). The second method also offer ability to separate the link and display title

```html

<div id='' ...... data-source='https://link'></div>

```

OR

```python

data['metadata']['source'] = {
    'title': 'Source Title',
    'link': 'https://link',
}

```

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Screenshots
untitled

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: hurumap/templates/profile/profile_detail.html [hunk 1]
```diff
@@ -873,10 +873,149 @@ <h1>Livestock</h1>
...
+                    return parentLinkSet;
+                }
+            })
+            // push the whole thing into the header box ... thingy
+            parentLinkContainer.html(parentLinksPrefix + parentLinkSets.join(', '));
+            $('body').trigger('glossaryUpdate', parentLinkContainer);
+        });
+}
+
+</script>
```
Comment (DavidLemayian) [inline hurumap/templates/profile/profile_detail.html]: This looks like a lot of JS though to have in the HTML. Is all of it really needed?

Comment (kilemensi) [reply hurumap/templates/profile/profile_detail.html]: Yes, it's a lot but it's needed. It's where Wazimap creates the charts so we need to override with our own options.

But, we could move it out of html into to `app.js` or something for better organization and just include it in the `{% block body_javascript_extra %}`.

Comment (DavidLemayian) [reply hurumap/templates/profile/profile_detail.html]: @kilemensi depends on how Wazimap implements this JS. Are we overriding an entire file or just a block?

Comment (kilemensi) [reply hurumap/templates/profile/profile_detail.html]: Just the ```{% block body_javascript_extra %}``` block @DavidLemayian.

Comment (DavidLemayian) [reply hurumap/templates/profile/profile_detail.html]: Cool. Then let's do it how Wazimap does it to avoid conflicts.

Comment (kilemensi) [reply hurumap/templates/profile/profile_detail.html]: Yeah, this is how [Wazimap](https://github.com/OpenUpSA/wazimap/blob/c39c44c420ff4eff360618d6dbfb821b95166edb/wazimap/templates/profile/profile_detail.html#L123) does it.

**Comment by DavidLemayian** [comment]
@kilemensi I guess resolve conflict and merge. :)

**Pull request merged**